### DISCUSSION TITLE
fastddsgen and ros2 gen msg, TypeSupport member m_typeSize id different, why?

### DISCUSSION THREAD

**Discussion by Eternity1987** [eProsima/Fast-DDS-Gen#427] [Support]
file Vec2Image1m.msg ：

shm_msgs/Image1m[2] images

fastddsgen:
fastddsgen -d {src_path} -de final  -I . -t {temp_path} -typeros2 -cs {idls_str}

![img_v3_02i3_eda8b31a-f570-4da3-9cd6-58404a93952g](https://github.com/user-attachments/assets/cde5f265-d220-4029-8ceb-333968ab0e13)

![img_v3_02i3_47995073-60e8-4e35-9a35-777fd1552bbg](https://github.com/user-attachments/assets/445f2fdf-9c11-491a-99d8-71a0576a6908)

ros2 gen:

![img_v3_02i3_d9a19fff-b384-47e2-bd5a-1fcce0526a4g](https://github.com/user-attachments/assets/6dcd4510-4771-4d8f-bb4e-8969b27bab47)

fastddsgen m_typeSize value is: 2098236 + 4
ros2 m_typeSize value is: 2098232 + 4

can you help me explain?

**Comment by richiware** [comment]
@Eternity1987 

*fastddsgen* calculates the maximum size that a type can occupy considering two encodings: XCDRv1 and XCDRv2. *ros2 gen* only calculates the maximum size for XCDRv1 encoding, which is the difference you see. Your type, by default, is \@appendable and 4 extra bytes are needed in XCDRv2 encoding.

Finally, I am moving this conversation to the Support discussion forum as it is not a proper bug in the application.

**Reply by Eternity1987** [reply]
> @Eternity1987
> 
> _fastddsgen_ calculates the maximum size that a type can occupy considering two encodings: XCDRv1 and XCDRv2. _ros2 gen_ only calculates the maximum size for XCDRv1 encoding, which is the difference you see. Your type, by default, is @appendable and 4 extra bytes are needed in XCDRv2 encoding.
> 
> Finally, I am moving this conversation to the Support discussion forum as it is not a proper bug in the application.


Thank you for your response to my question, but I want it to only generate "-cdr v1", with the size remaining the same.
![image](https://github.com/user-attachments/assets/33c181c8-1394-432c-b1d3-16ac4de29b6a)
![image](https://github.com/user-attachments/assets/81cad749-b82f-4723-a1e9-a4e508351c12)

full command: 
 fastddsgen -d {src_path} -de final -I . -t {temp_path} -typeros2 -cs {idls_str} -cdr v1

**Comment by richiware** [comment]
The `-cdr` argument selects the version of *FastCDR* against which the generated code should be compiled. But the calculation of the maximum size of a type is independent of this. It always calculates the larger value needed to store your encoded type in XCDRv1 or XCDRv2.
